File: MiniDOMAsteriskLib/CallManagers/Events/DialEvent.vb
Imports System.Net.Sockets
Imports System.Threading
Imports System.Net

Namespace CallManagers


    Public Class DialEvent
        Inherits AsteriskEvent




        Public Sub New()
        End Sub

        Public Sub New(ByVal server As AsteriskCallManager, ByVal rows() As String)
            MyBase.New(server, rows)
        End Sub

        Public Sub New(ByVal server As AsteriskCallManager, ByVal e As AsteriskEvent)
            MyBase.New(server, e)
        End Sub

        Public ReadOnly Property CallerIDNumber As String
            Get
                Return Me.GetAttribute("CALLERIDNUM")
            End Get
        End Property

        Public ReadOnly Property Channel As String
            Get
                Return Me.GetAttribute("CHANNEL")
            End Get
        End Property

        Public ReadOnly Property ConnectedLineName As String
            Get
                Return Me.GetAttribute("CONNECTEDLINENAME")
            End Get
        End Property

        Public ReadOnly Property ConnectedLineNum As String
            Get
                Return Me.GetAttribute("CONNECTEDLINENUM")
            End Get
        End Property

        Public ReadOnly Property Destination As String
            Get
                Return Me.GetAttribute("DESTINATION")
            End Get
        End Property

        Public ReadOnly Property DestUniqueID As String
            Get
                Return Me.GetAttribute("DESTUNIQUEID")
            End Get
        End Property

        Public ReadOnly Property DialString As String
            Get
                Return Me.GetAttribute("DIALSTRING")
            End Get
        End Property

        Public ReadOnly Property SubEvent As String
            Get
                Return Me.GetAttribute("SUBEVENT")
            End Get
        End Property

        Public ReadOnly Property UniqueID As String
            Get
                Return Me.GetAttribute("UNIQUEID")
            End Get
        End Property

    End Class


End Namespace

File: MiniDOMAsteriskLib/CallManagers/Events/DialEvent.cs

namespace minidom.CallManagers
{
    public class DialEvent : AsteriskEvent
    {
        public DialEvent()
        {
        }

        public DialEvent(AsteriskCallManager server, string[] rows) : base(server, rows)
        {
        }

        public DialEvent(AsteriskCallManager server, AsteriskEvent e) : base(server, e)
        {
        }

        public string CallerIDNumber
        {
            get
            {
                return GetAttribute("CALLERIDNUM");
            }
        }

        public string Channel
        {
            get
            {
                return GetAttribute("CHANNEL");
            }
        }

        public string ConnectedLineName
        {
            get
            {
                return GetAttribute("CONNECTEDLINENAME");
            }
        }

        public string ConnectedLineNum
        {
            get
            {
                return GetAttribute("CONNECTEDLINENUM");
            }
        }

        public string Destination
        {
            get
            {
                return GetAttribute("DESTINATION");
            }
        }

        public string DestUniqueID
        {
            get
            {
                return GetAttribute("DESTUNIQUEID");
            }
        }

        public string DialString
        {
            get
            {
                return GetAttribute("DIALSTRING");
            }
        }

        public string SubEvent
        {
            get
            {
                return GetAttribute("SUBEVENT");
            }
        }

        public string UniqueID
        {
            get
            {
                return GetAttribute("UNIQUEID");
            }
        }
    }
}